packages/backend/src/modules/user/user.repository.ts

```diff
@@ -1,49 +1,74 @@
 /* eslint-disable no-underscore-dangle */
-import { CreateUserCommand, UpdateUserCommand } from 'shared'
+import {
+  CannotCreateDocumentError,
+  CannotDeleteDocumentError,
+  CannotFindDocumentError,
+  CannotUpdateDocumentError,
+  CreateUserCommand,
+  UpdateUserCommand
+} from 'shared'
 import mongoose from 'mongoose'
 import { User } from './user'
-import userModel from './user.model'
+import userModel, { UserDocument, UserModel } from './user.model'
+import userMapper, { UserMapper } from './user.mapper'
 
 export class UserRepository {
-  private readonly _model: mongoose.Model<User>
+  private readonly _model: UserModel
+  private readonly _mapper: UserMapper
 
-  constructor(model: mongoose.Model<User>) {
+  constructor(model: UserModel, mapper: UserMapper) {
     this._model = model
+    this._mapper = mapper
   }
 
-  async create(dto: CreateUserCommand): Promise<User> {
-    const newUser: User = await this._model.create(dto)
-    return newUser
+  async create(command: CreateUserCommand): Promise<User> {
+    try {
+      const newUser: UserDocument = await this._model.create(command)
+      return this._mapper.mapToDomain(newUser)
+    } catch (e) {
+      throw new CannotCreateDocumentError('User', command)
+    }
   }
 
-  async updateOneById(id: string, dto: UpdateUserCommand): Promise<User> {
-    const patchedUser: User | null = await this._model.findByIdAndUpdate(id, dto, { new: true })
-    if (patchedUser === null) {
-      throw new Error()
+  async updateOneById(id: string, command: UpdateUserCommand): Promise<User> {
```

Comment:
there is no need to use UpdateUserCommand, we should use here UserDTO instead

Reply:
imho we should remove UpdateUserCommand because it is almost the same as UserDTO

Reply:
imho we should remove UpdateUserCommand because it is almost the same as UserDTO

Reply:
I think the commands are fine, in `UserDTO` we have the `id` field present and almost all fields are required, while in `UpdateUserCommand` all fields are optiona. In `CreateUserCommand` the `id` field is not present.

Reply:
Generally it's a good approach, but in our case it will be almost 1:1 as UserDTO where most of field will be also optional 

It's not wrong, if you want you can leave it as it is. It's just redundant